tests/unit/lms/services/async_oauth_http_test.py

```diff
@@ -0,0 +1,81 @@
+import asyncio
+from unittest.mock import sentinel
+
+import httpretty
+import pytest
+from aiohttp import TooManyRedirects
+from aioresponses import CallbackResult, aioresponses
+
+from lms.services import ExternalRequestError
+from lms.services.async_oauth_http import AsyncOAuthHTTPService, factory
+
+
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
+class TestAsyncOAuthHTTPService:
+    def test_get_success(
+        self, svc, urls, with_successful_responses, oauth2_token_service
+    ):
+        responses = svc.get(urls)
+
+        for i, url in enumerate(urls):
+            assert responses[i].status == 200
+
+            # Order is kept from the URLs param
+            assert responses[i].headers["url"] == url
+
+        for request in with_successful_responses.requests.values():
+            request_kwargs = request[0].kwargs
+
+            assert request_kwargs["timeout"] == 10
+            assert (
+                request_kwargs["headers"]["Authorization"]
+                == f"Bearer {oauth2_token_service.get().access_token}"
+            )
+
+    @pytest.mark.usefixtures("with_one_failed_response")
+    def test_get_with_failure(self, svc, urls):
+        with pytest.raises(ExternalRequestError):
+            svc.get(urls)
+
+    @pytest.fixture
+    def with_successful_responses(self, urls):
+        with aioresponses() as m:
+            for url in urls:
+                m.get(url, headers=dict(url=url))
+
+            yield m
+
+    @pytest.fixture
+    def with_one_failed_response(self, urls):
+        async def slow_response_callback(_url, **_kwargs):
+            await asyncio.sleep(60 * 60 * 24)
+            return CallbackResult(status=200)
+
+        with aioresponses() as m:
+            for url in urls[:-1]:
+                m.get(url, callback=slow_response_callback)
+            m.get(urls[-1], exception=TooManyRedirects("info", "history"))
+
+            yield m
+
+    @pytest.fixture
+    def urls(self):
+        """Return the URLs that we'll be sending test requests to."""
+        return ["https://example.com/example", "https://example.com/another"]
+
+    @pytest.fixture
+    def svc(self, oauth2_token_service):
+        return AsyncOAuthHTTPService(oauth2_token_service)
+
+    @pytest.fixture(autouse=True)
+    def unhttpretty(self):
```

Comment:
```suggestion
    def httpunpretty(self):
```

:)

Although I think you can just override the `httpretty_` fixture to disable it for tests in this class:

```python
class TestAsyncOAuthHTTPService:
    ...

    @pytest.fixture(autouse=True)
    def httpretty_(self):
        pass
```

Is this necessary though? The tests seem to pass without it

Reply:
:thinking: This was causing an error when I tested it locally but now it works with it, removed.